src/librustc_trans/back/write.rs

```diff
@@ -551,7 +552,8 @@ unsafe fn optimize_and_codegen(cgcx: &CodegenContext,
         if config.emit_obj {
             let path = output_names.with_extension(&format!("{}.o", name_extra));
             with_codegen(tm, llmod, config.no_builtins, |cpm| {
-                write_output_file(cgcx.handler, tm, cpm, llmod, &path, llvm::ObjectFileType);
+                write_output_file(cgcx.handler, tm, cpm, llmod, &path, config.emit_asm,
```

Comment:
Could you add a comment here indicating why we're skipping codegen in the case that assembly is being emitted? (a pointer to the C++ file is fine)